app/src/main/java/com/keylesspalace/tusky/service/SendStatusService.kt

```diff
@@ -244,18 +260,22 @@ class SendStatusService : Service(), Injectable {
                 notificationManager.cancel(statusId)
             }, { throwable ->
                 Log.w(TAG, "failed sending status", throwable)
-                if (throwable is HttpException) {
-                    // the server refused to accept the status, save status & show error message
-                    failSending(statusId)
-                } else {
-                    // a network problem occurred, let's retry sending the status
-                    retrySending(statusId)
-                }
+                failOrRetry(throwable, statusId)
             })
             stopSelfWhenDone()
         }
     }
 
+    private suspend fun failOrRetry(throwable: Throwable, statusId: Int) {
+        if (throwable is HttpException) {
+            // the server refused to accept, save status & show error message
+            failSending(statusId)
+        } else {
+            // a network problem occurred, let's retry sending the status
+            retrySending(statusId)
```

Comment:
This would retry indefinitely? (retries are counted but I don't see a cancel?)

What kind of exceptions are network problems?

Reply:
Yes this retries until it is cancelled, the whole idea behind this construct is that status gets send once the device is connected again.

Network problem is everything that causes the call to fail before a server response (successfull or unsuccessfull) is received.